client/allocrunner/taskrunner/task_runner.go

```diff
@@ -388,6 +388,28 @@ func (tr *TaskRunner) initLabels() {
 	}
 }
 
+// Mark a task as failed and not to run.  Aimed to be invoked when alloc runner
+// prestart hooks failed.
+// Should never be called with Run().
+func (tr *TaskRunner) MarkFailedDead(reason string) {
```

Comment:
Here, we introduce another function to call instead of `Run()` breaking an invariant; but I had a very hard time rationalizing reusing `Run()` when we never want to call any of the logic there and we would need to signal that the task should fail.

I felt that `MarkFailedDead` is a reasonable compromise that does the bare minimum to mark task as failed.